utils/TestUtils.ts

```diff
@@ -50,33 +50,11 @@ export class ValkeyCluster {
         this.version = version;
     }
 
-    private static async detectVersion(): Promise<string> {
-        return new Promise<string>((resolve, reject) => {
-            const extractVersion = (stdout: string): string =>
-                stdout.split("v=")[1].split(" ")[0];
-
-            // First, try with `valkey-server -v`
-            exec("valkey-server -v", (error, stdout) => {
-                if (error) {
-                    // If `valkey-server` fails, try `redis-server -v`
-                    exec("redis-server -v", (error, stdout) => {
-                        if (error) {
-                            reject(error);
-                        } else {
-                            resolve(extractVersion(stdout));
-                        }
-                    });
-                } else {
-                    resolve(extractVersion(stdout));
-                }
-            });
-        });
-    }
-
     public static createCluster(
         cluster_mode: boolean,
         shardCount: number,
         replicaCount: number,
+        getVersionCallback: (addresses: [string, number][]) => Promise<string>,
```

Comment:
The callback still have 1 argument, should have 2

Reply:
Updated.